examples/github-notification-filters/app.tsx

```diff
@@ -0,0 +1,212 @@
+/** @jsx h */
+import { autocomplete } from '@algolia/autocomplete-js';
+import { createTagsPlugin, Tag } from '@algolia/autocomplete-plugin-tags';
+import { h, render } from 'preact';
+import qs from 'qs';
+
+import {
+  FilterHeader,
+  PanelLayout,
+  PostfixItem,
+  PrefixItem,
+  QueryItem,
+  TagItem,
+} from './components';
+import { items } from './items';
+import { AutocompleteItem, NotificationFilter } from './types';
+import { getAlltags, groupBy, splitQuery } from './utils';
+
+import '@algolia/autocomplete-theme-classic';
+
+const queryParameters = qs.parse(location.search, { ignoreQueryPrefix: true });
+
+const tagsPlugin = createTagsPlugin<AutocompleteItem, NotificationFilter>({
+  initialTags: Object.entries(queryParameters).flatMap(([token, values]) =>
+    (values as string[]).map((value) => ({
+      token,
+      value,
+      label: `${token}:${value}`,
+    }))
+  ),
+  getTagsSubscribers() {
+    return [
+      {
+        sourceId: 'postfixes',
+        getTag({ item }) {
+          return {
+            label: `${item.token}:${item.label}`,
+            token: item.token,
+            value: item.label,
+          };
+        },
+      },
+    ];
+  },
+  transformSource() {
+    return undefined;
+  },
+  onChange({ tags }) {
+    onTagsChange(tags);
+  },
+});
+
+autocomplete<AutocompleteItem>({
+  container: '#autocomplete',
+  placeholder: 'Filter notifications',
+  openOnFocus: true,
+  defaultActiveItemId: 0,
+  plugins: [tagsPlugin],
+  detachedMediaQuery: 'none',
+  getSources({ query, state }) {
+    const [prefix, postfix] = splitQuery(query);
+    const prefixes = items.filter(({ token }) => token.startsWith(prefix));
+
+    const allTags = getAlltags(state.context.tagsPlugin?.tags || [], query);
+    const showQuerySource = allTags.length > 0 && prefixes.length > 0;
+    const showPrefixesSource = typeof postfix !== 'string';
+    const showPostfixesSource = typeof postfix === 'string';
+
+    const querySource = {
+      sourceId: 'query',
+      onSelect() {
+        const params = Object.fromEntries(
+          Object.entries(
+            groupBy(allTags, ({ token }) => token)
+          ).map(([key, entries]) => [key, entries.map(({ value }) => value)])
+        );
+
+        window.location.assign(
+          [window.location.origin, qs.stringify(params)].join('?')
+        );
+      },
+      getItems() {
+        return [{}] as Array<Tag<NotificationFilter>>;
+      },
+      templates: {
+        item() {
+          return (
+            <QueryItem query={allTags.map(({ label }) => label).join(' ')} />
+          );
+        },
+      },
+    };
+    const prefixesSource = {
+      sourceId: 'prefixes',
+      onSelect({ item, setQuery, setIsOpen, refresh }) {
+        setQuery(`${item.token}:`);
+        setIsOpen(true);
+
+        refresh();
+      },
+      getItems() {
+        if (query.length === 0) {
+          return items;
```

Comment:
should this maybe be called `filters`, `tags`, `options`? items is confusing with hits imo. Maybe even `possiblePrefixes`

Reply:
I wanted to avoid `tags` because it's something else, and `filters` because `filters.filter(/* ... */)` is weird, also there's a `filtered` variable at some point and that makes it even weirder 😁 And `options` is usually for function parameters (when an optional object).

The name `items` seemed to fit because it's what `getItems` returns, it is what then becomes hits (once returned).